### PR TITLE
Breadcrumbs Enabled In All of Spark

### LINKED ISSUE #127: Add breadcrumbs to all modules - New Feature
New Feature - Expand current implementation to all modules.

### Overview of the problem
This is about the Spark **Users module**

### Description

In order to enable breadcrumbs in all modules we must:
1. Move the init command from camp-routes to main-routes:
`app.use(breadcrumbs.init());` - camps_routes.js line 9

2. Declare Home page - this will store a persistent home page throughout the app:
`app.use(breadcrumbs.setHome({
  name: 'home',
  url: '/' + req.params.lng + '/'
}));`

3. To any new route add to the  req a breadcrumb object or object array. Each object is made of a name and a url. The name should be the translation string. For examples see camps_routes (if not merged see PR #124 )

4. Add page names to locals.

5. In case we implement across the board - remove 'if' condition from page.jade (line 45).

6. Test links ;)

### PR DESCRIPTION
**Pull request by guywarburg**
### Proposed solution
Closes #127 - Added breadcrumbs to whole website.

### Tradeoffs
Accidentally branched out of a local branch so made some extra changes to revert those.

### Testing Done
Tests pass.

### CODE REVIEW

**Comment by guywarburg** [comment]
I would like to us the feature:
`breadcrumbs.setHome()` so I can clean up the code.
However, is this is set up before any request is made I don't have access to the language.
I wanted to ask if there's another place where I can access the user's language - some sort of cookie or another way?

In used `req.params.lng` but as we aren't in the scope of a specific request I can't access that and looking for an alternative.

**Review by OriHoch — APPROVED** [review]
nice!

**Comment by OriHoch** [comment]
@guywarburg - about your question regarding accessing req.params.lng - 

I'm not familiar with `breadcrumbs.setHome()` - but, IIUC - this function runs in a scope that doesn't have access to req object. 
Does the req object exist while it's running but it just doesn't have access to it? or - it doesn't exist yet at all?

I think the best way is to make sure req object exists while this function runs and also make sure it has access to it (worst case scenario - via a global variable)

**Comment by guywarburg** [comment]
@OriHoch - I tried to initiate the `setHome()` within the first `.get()` call to the server but it doesn't work there (not sure why). It must be set-up immediately after `breadcrumbs.init()` (main_routes.js line 21). This is before any requests have been made so we can't specify a language (or at least as far as I know).
In second thought, maybe the better idea is to set the default language according to a cookie and not just hard coded Hebrew. Once this is set up the home breadcrumb url will be set simply to '/' and the routes will add the language according to the user's choice.

**Comment by natekonimbo** [comment]
awesome! merged with master: latest changes, but notice that:

```
15/03/17 06:18:14.127 INFO app.js : GET /he/admin 302 60.832 ms - 70
15/03/17 06:18:14.651 INFO app.js : GET /he/login?r=/ 304 517.716 ms - -
```

can you fix ?

**Comment by guywarburg** [comment]
All tests passed locally - this is odd

**Comment by LeonFedotov** [comment]
@guywarburg it seems like part of the implementation is there on master, is this pr still relevant? if so please merge from master.

**Comment by guywarburg** [comment]
@LeonFedotov - the idea was to add the breadcrumbs across the board (to all pages).
I'll fix it soon, it's the 2nd time it isn't merged in time so it creates conflicts...

**Comment by LeonFedotov** [comment]
Yes i see the conflicts, I went over them and there are some stuff that has
been already added to master, maybe you can try to merge master to your
branch and see if the new changes fit your solution :)

File: views/pages/camps/camp.jade [hunk 1]
```diff
@@ -14,6 +14,44 @@ block content
                 .clearfix
                 .desc
                     h5 #{t('camps:camps.desc')}
+<<<<<<< HEAD
```
Comment (LeonFedotov) [inline views/pages/camps/camp.jade]: hey @guywarburg i think you got a situation here

File: routes/camps_routes.js [hunk 2]
```diff
@@ -125,8 +159,11 @@ module.exports = function (app, passport) {
             }).fetch().then((user) => {
                 res.render('pages/camps/camp', {
                     user: req.user,
+                    userLoggedIn: req.user.hasRole('logged in'),
                     id: req.params.id,
                     camp: camp.toJSON(),
+                    details: camp.related('details').toJSON(),
```
Comment (LeonFedotov) [inline routes/camps_routes.js]: this is a duplicate key

File: routes/camps_routes.js [hunk 3]
```diff
@@ -153,6 +206,9 @@ module.exports = function (app, passport) {
             }
             res.render('pages/camps/edit', {
                 user: req.user,
+                camp: camp.toJSON(),
```
Comment (LeonFedotov) [inline routes/camps_routes.js]: this is another duplicate key

**Review by LeonFedotov — CHANGES_REQUESTED** [review]

**Comment by guywarburg** [comment]
@LeonFedotov - fixed duplicates and improved breadcrumbs for new flow

**Review by LeonFedotov — APPROVED** [review]
looks good

**Comment by amotenko** [comment]
@guywarburg it seems like there are minor conflicts that are not your fault, can you please sort them so I'll be able to merge

**Comment by guywarburg** [comment]
@amotenko - fixed please merge

**Pull request merged**